data_structures/ringbuffer.c

```diff
@@ -0,0 +1,129 @@
+/**
+ * Speedster3030, 2025
+ * Ring Buffer/Circular Queue structure implementation.
+ */
+
+#include<stdio.h>
+#include<stdlib.h>
+
+//ring Buffer/Circular Queue struct
+typedef struct
+{
+        int* arr;
+        int count;
+        int head;
+        int tail;
+        int max;
+}ringBuffer;
+
+//Initialize a new ring buffer
+ringBuffer* new_ringBuffer(int size)
+{
+        ringBuffer* rb=malloc(sizeof(ringBuffer));
+
+        rb->max=size;
+        rb->arr=malloc(size*sizeof(int));
+        rb->count=0;
+        rb->head=size-1;
+        rb->tail=size-1;
+
+        return rb;
+}
+
+//free the structure from memory
+void delete_Buffer(ringBuffer **rb)
+{
+        free((*rb)->arr);
+        free(*rb);
+        *rb=NULL;
+}
+
+//Put a value in the buffer
+int enqueue(ringBuffer *rb,int n)
+{
+        if(rb->count==rb->max)
+        {
+                return -1;
+        }
+
+        rb->arr[rb->tail]=n;
+        rb->tail--;
+        if(rb->tail==-1)
+        {
+                rb->tail=rb->max-1;
+        }
```

Comment:
Consider  more conventional style for circular queues:

```suggestion
        rb->arr[rb->tail] = n;
        rb->tail = (rb->tail + 1) % rb->max;
```